### PR TITLE
EVG-19942: Add create/copy distro interactions

### PR DESCRIPTION
**Pull request by sophstad**
EVG-19942

### Description

- Add menu and modals for creating and copying distros
- Only show button if user has create distro permissions
- Rename user permissions resolver to user project settings permission resolver to more clearly distinguish between the distro permissions resolver
- The ticket mentions adding a warning state if the distro was created with an error, but I designed the resolvers to fail if an error is encountered, so this pattern was not implemented. It makes more sense for projects where unexpected conflicts could occur when e.g. creating a project that would introduce commit queue collisions; distros are simpler.

### Screenshots

image
image
image

### Testing

- Unit test modals
- Add integration tests that can be isolated once a user can delete a distro from the UI (EVG-19943)

### CODE REVIEW

File: src/pages/distroSettings/CopyModal.test.tsx [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+const distroMock: ApolloMock<DistroQuery, DistroQueryVariables> = {
+  request: {
+    query: DISTRO,
+    variables: {
+      distroId: newDistroId,
+    },
+  },
+  result: {
+    data: {
+      distro: { name: newDistroId },
```
Comment (minnakt) [inline src/pages/distroSettings/CopyModal.test.tsx:167]: nit. add typename here

File: src/pages/distroSettings/CreateModal.test.tsx [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+const distroMock: ApolloMock<DistroQuery, DistroQueryVariables> = {
+  request: {
+    query: DISTRO,
+    variables: {
+      distroId: newDistroId,
+    },
+  },
+  result: {
+    data: {
+      distro: { name: newDistroId },
```
Comment (minnakt) [inline src/pages/distroSettings/CreateModal.test.tsx:149]: nit. add typename here

File: src/pages/distroSettings/CreateModal.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+  const onConfirm = () => {
+    createDistro({
+      variables: {
+        opts: {
+          newDistroId: formState.newDistroId,
+        },
+      },
+    });
+    sendEvent({ name: "Create new distro" });
```
Comment (minnakt) [inline src/pages/distroSettings/CreateModal.tsx:52]: maybe we want to include the name of the new distro!

File: src/pages/distroSettings/NewDistroButton.tsx [hunk 4]
```diff
@@ -0,0 +1,76 @@
+import { useState } from "react";
```
Comment (minnakt) [inline src/pages/distroSettings/NewDistroButton.tsx:1]: idk how you feel about this but I ended up putting `DistroSelect` in its own folder — does it make sense to put these components in a folder too? or you can un-folder-ify the `DistroSelect` if that's better

File: src/pages/distroSettings/NewDistroButton.tsx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+  return (
+    <>
+      <Menu
+        data-cy="new-distro-menu"
+        open={menuOpen}
+        popoverZIndex={zIndex.popover}
+        setOpen={setMenuOpen}
+        trigger={
+          <Button
```
Comment (minnakt) [inline src/pages/distroSettings/NewDistroButton.tsx:38]: should this button be green? :0 (based on the designs)

**Review by minnakt — APPROVED** [review]
looks great! 🤩

**Comment by sophstad** [comment]
Unit test failure appears unrelated and tracked by EVG-20570

**Pull request merged**